### DISCUSSION TITLE
Changing color of model elements

### DISCUSSION THREAD

**Discussion by li-arditi** [iTwin/itwinjs-core#3290] [Q&A] [answered]
Hello, so I have an iTwin Viewer React App working using the iTwin accreditation course as a guide. Now I am trying to make specific elements in the model be specific colors. Since the elements don't seem to have many properties, I think I will need to provide arrays of ECInstanceIDs obtained based on separate data. Another big part is that the model is a 4D model so there is a time scrubber on the bottom that shows the evolution of the project/model. I'd like that data to be incorporated such that the colors will change depending on time. I'm new to this iTwin/iModel stuff so I really don't know where to start. I was potentially thinking the Emphasize Elements sample could be helpful for changing the color of the elements, but other than that I'm a bit lost. I've been watching the accreditation course videos and looking through the online examples but haven't been able to find anything to give me somewhere to start. Any guidance or suggestions or resources or examples that I missed would be greatly appreciated.

**Comment by pmconne** [comment, marked as answer]
You can use [appearance overrides](https://www.itwinjs.org/learning/display/symbologyoverrides/) to override the colors of elements.
You can use a [render timeline script](https://www.itwinjs.org/learning/display/timelineanimation/) to override their colors over time.
These articles contain links to interactive samples and more detailed API documentation. Feel free to follow up if you still have questions after perusing them.

**Reply by li-arditi** [reply]
Thank you for such a quick response! For the timeline animation part, is there a way to make it so the timeline doesn't progress automatically? For example, if I have this 
![timescrubber](https://user-images.githubusercontent.com/60494701/156431464-43872a5b-33e6-4695-9262-fcc5486af652.png)
then can I have it so the colors change based on moving the sliding bar? Or does this still fall under Timeline Animation

**Reply by pmconne** [reply]
I believe that is how the timeline component already works - if you pause it, you can drag the scrubber to the desired time point and see the view update accordingly.
Is that not working for you? (@bsteinbk wrote that component - I am only vaguely familiar with it).

**Comment by bsteinbk** [comment]
As the timeline scrubber moves it calls either
	`viewport.analysisFraction = animationFraction` for an AnalysisStyle 
or
	`viewport.timePoint = script.duration.fractionToPoint(animationFraction)` if a scheduleScript is defined
